examples/filter2/main.go

```diff
@@ -0,0 +1,156 @@
+package main
+
+import (
+	"context"
+	"crypto/rand"
+	"encoding/hex"
+	"fmt"
+	"net"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	"github.com/ethereum/go-ethereum/crypto"
+	logging "github.com/ipfs/go-log"
+	"github.com/status-im/go-waku/waku/v2/node"
+	"github.com/status-im/go-waku/waku/v2/protocol/pb"
+)
+
+var log = logging.Logger("filter2")
+
+const pubSubTopic = ""
+const contentTopic = "test"
+
+func main() {
+	lvl, err := logging.LevelFromString("info")
+	if err != nil {
+		panic(err)
+	}
+	logging.SetAllLoggers(lvl)
+
+	hostAddr1, _ := net.ResolveTCPAddr("tcp", fmt.Sprint("0.0.0.0:60000"))
+	hostAddr2, _ := net.ResolveTCPAddr("tcp", fmt.Sprint("0.0.0.0:60002"))
+
+	key1, err := randomHex(32)
+	if err != nil {
+		log.Error("Could not generate random key")
+		return
+	}
+	prvKey1, err := crypto.HexToECDSA(key1)
+
+	key2, err := randomHex(32)
+	if err != nil {
+		log.Error("Could not generate random key")
+		return
+	}
+	prvKey2, err := crypto.HexToECDSA(key2)
+
+	ctx := context.Background()
+
+	fullNode, err := node.New(ctx,
+		node.WithPrivateKey(prvKey1),
+		node.WithHostAddress([]net.Addr{hostAddr1}),
+		node.WithWakuRelay(),
+		node.WithWakuFilter(),
+	)
+
+	lightNode, err := node.New(ctx,
+		node.WithPrivateKey(prvKey2),
+		node.WithHostAddress([]net.Addr{hostAddr2}),
+		node.WithWakuRelay(),
+		node.WithWakuFilter(),
+	)
+
+	_, err = lightNode.AddFilterPeer(fullNode.ListenAddresses()[0])
+	if err != nil {
+		log.Info("Error adding filter peer on light node ", err)
+	}
+	//
+	// Setup filter
+	//
+
+	// Send FilterRequest from light node to full node
+	filterRequest := pb.FilterRequest{
+		ContentFilters: []*pb.FilterRequest_ContentFilter{&pb.FilterRequest_ContentFilter{ContentTopic: contentTopic}},
+		Topic:          pubSubTopic,
+		Subscribe:      true,
+	}
+	filterHandler := func(msg pb.WakuMessage) {
+		log.Info("Light node received msg, ", string(msg.Payload))
+	}
+	lightNode.SubscribeFilter(filterRequest, filterHandler)
```

Comment:
What do you think of using channels instead of a the filterHandler callback?

Reply:
Yep, good idea, modified it to channels.